### PR TITLE
Add last note padding test case using softmax

### PR DESCRIPTION
**Pull request by AaronDavidNewman**
This addresses issue 1370.  Added a test case to formatter test cases to demonstrate softmax and how it affects measures with extra whitespace.  I don't think there is any other action we need to take, but this seems to come up a lot.

image

### CODE REVIEW

**Comment by rvilarl** [comment]
@AaronDavidNewman shouldn't we change the default from 100 to 5 as well?

File: tests/formatter_tests.ts [hunk 1]
```diff
@@ -147,6 +149,48 @@ function rightJustify(options: TestOptions): void {
...
+    voices.push(score.voice(notes).setMode(2));
+    notes = [];
+    stave = system.addStave({ voices: voices });
+    stave.addClef('bass');
+    stave.addTimeSignature('2/4');
+    voices = [];
+    f.draw();
+    y += 100;
+  };
+  draw(100);
```
Comment (rvilarl) [inline tests/formatter_tests.ts:186]: It would be good to add a text indicating which softmax value was used.

File: tests/formatter_tests.ts [hunk 2]
```diff
@@ -119,7 +121,7 @@ function buildTickContexts(): void {
 }
 
 function rightJustify(options: TestOptions): void {
-  const f = VexFlowTests.makeFactory(options, 1200, 300);
+  const f = VexFlowTests.makeFactory(options, 1200, 150);
```
Comment (rvilarl) [inline tests/formatter_tests.ts:124]: why did you change this value?

File: tests/formatter_tests.ts [hunk 3]
```diff
@@ -132,7 +134,7 @@ function rightJustify(options: TestOptions): void {
   const renderTest = (time: VoiceTime, n: number, duration: string, duration2: string, x: number, width: number) => {
     const formatter = f.Formatter();
 
-    const stave = f.Stave({ x, y: 40, width });
+    const stave = f.Stave({ x, y: 20, width });
```
Comment (rvilarl) [inline tests/formatter_tests.ts:137]: Why did you change this value?

**Review by rvilarl — CHANGES_REQUESTED** [review]

File: tests/formatter_tests.ts (hunk 2 above)
Comment (AaronDavidNewman) [reply tests/formatter_tests.ts:124]: Trying to conserve vertical space.  Will it mess with the visual regression?

**Comment by AaronDavidNewman** [comment]
> 

That would change _all_ the test cases. Do we want to do that?

I'm also open to making it easier to set the global softmax.  It seems like it is defined in 2 places.  It seemse like it should be defined in one place, either be in tables, or in the font metrics.

**Comment by AaronDavidNewman** [comment]
image

image

File: tests/formatter_tests.ts (hunk 2 above)
Comment (rvilarl) [reply tests/formatter_tests.ts:124]: That is ok I thought that it might have been overseen.

**Review by rvilarl — APPROVED** [review]

**Comment by rvilarl** [comment]
> > shouldn't we change the default from 100 to 5 as well?
> 
> That would change _all_ the test cases. Do we want to do that?
> 
> I'm also open to making it easier to set the global softmax. It seems like it is defined in 2 places. It seemse like it should be defined in one place, either be in tables, or in the font metrics.

I would go for that, @0xfe what do you think?

**Comment by AaronDavidNewman** [comment]
I am going to cancel this pull request and resubmit because I screwed up my local branches.
